### DISCUSSION TITLE
/api/v2/firewall/states could have a DELETE method to delete ALL states

### DISCUSSION THREAD

**Discussion by HovKlan-DH** [jaredhendrickson13/pfsense-api#527] [Ideas]
It would be nice if it was possible to reset ALL states in one go, and I think this could easily be archived by having a DELETE method for this endpoint? :-)

It is possible to delete all states via the normal UI from **Diagnostics** > **States** > **Reset States** > **Reset the firewall state table**

**Comment by jaredhendrickson13** [comment]
This is related to #494 and is already present in the `next_minor` branch (will become v2.1.0). Starting with v2.1.0 you'll be able to make DELETE requests to certain plural endpoints which will allow you delete objects that match a given query instead of requiring an individual ID. This would allow you to delete all objects as well. This will become the recommended method of deleting firewall states since DELETE /api/v2/firewall/state can be inconsistent.

**Comment by HovKlan-DH** [comment]
Sounds great and I will then close this topic.
